### PR TITLE
Hacks in browserify + google-closure-compiler bundling step.

### LINKED ISSUE #13: ENHANCEMENT: Add Browserify/GCC bundler
### Description
Add new bundler to the `usng2-js` build file, swapping out the webpack bundler. This relates to #11.

The updated build file should be simplified and run slightly faster as the webpack step will no longer be performed only for the output to be deleted.

[Description of the issue]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

#### Expected behavior:
[What you expect to happen]

#### Actual behavior:
[What actually happens]

#### Reproduces how often:
[What percentage of the time does it reproduce?]

### Version(s)
[Java library] or [Javascript library] version.

### Additional Information
Any additional information, configuration or data that might be necessary
to reproduce the issue.

### PR DESCRIPTION
**Pull request by coyotesqrl**
### Requirements
* Filling out the template is required. Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

This change uses browserify + google-closure-compiler to bundle and eliminate dead code from the final output file. The resulting javascript is UMD and has no external dependencies.

The ideal solution would add a new `bundler` implementation to the `KotlinFrontendExtension` and remove the `webpack` bundle plugin; however, there is an issue with the `npm-index` task in KFE right now that needs to be resolved before that can even be attempted. While this is kludgy, the output is correct.

### Alternate Designs

As mentioned above, a new bundler would be ideal.

### Benefits

This produces an npm artifact that works in both the browser and node.

### Possible Drawbacks

The final output file size is still rather large at ~660K.

In an environment with many Kotlin-derived JavaScript libraries, it would be preferable to exclude the Kotlin runtime in the output bundle and amortize the cost of that across multiple consumers.

### Checklist:
- [ ] Common Kotlin Unit Tests Added/Modified
- [ ] Javascript Mocha Tests Added to `js-test` module

### Applicable Issues

Fixes: #13

### CODE REVIEW

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

File: settings.gradle.kts [hunk 1]
```diff
@@ -7,18 +7,25 @@ or http://en.wikipedia.org/wiki/MIT_License
...
-                useModule("org.jetbrains.kotlin:kotlin-gradle-plugin:${requested.version}")
+            when (requested.id.id) {
+                "kotlin-platform-common" ->
+                    useModule("org.jetbrains.kotlin:kotlin-gradle-plugin:${requested.version}")
+                "kotlin-platform-jvm" ->
+                    useModule("org.jetbrains.kotlin:kotlin-gradle-plugin:${requested.version}")
+                "kotlin-platform-js" ->
+                    useModule("org.jetbrains.kotlin:kotlin-gradle-plugin:${requested.version}")
+                "kotlin-dce-js" ->
+                    useModule("org.jetbrains.kotlin:kotlin-gradle-plugin:${requested.version}")
```
Comment (brjeter) [inline settings.gradle.kts:26]: ✏️ You could technically combine all the identical branches into one, with comma-delimited string values.

**Review by brjeter — APPROVED** [review]

File: usng2-js/build.gradle.kts [hunk 2]
```diff
@@ -43,16 +41,25 @@ dependencies {
 configure<KotlinFrontendExtension> {
     downloadNodeJsVersion = "latest"
     sourceMaps = true
+    define("PRODUCTION", true)
```
Comment (AzGoalie) [inline usng2-js/build.gradle.kts:44]: ✏️ Probably doesn't matter at this size, but do we wan't to build production every time?

File: usng2-js/build.gradle.kts [hunk 3]
```diff
@@ -72,25 +79,75 @@ tasks {
...
             }
-            into(outDir)
         }
+    }
+
+    private fun copyJs(outDir: File) {
         project.copy {
-            from("${project.buildDir}/package.json")
+            from("${project.buildDir}/bundle/gcc-output.js") {
+                rename("gcc-output", "usng2.main")
```
Comment (AzGoalie) [inline usng2-js/build.gradle.kts]: ✏️ Would it be worth moving the final name into a variable?

**Review by AzGoalie — APPROVED** [review]

File: settings.gradle.kts (hunk 1 above)
Comment (coyotesqrl) [reply settings.gradle.kts:26]: Agreed. I'm being a bit overly verbose here and there may not be any value to it. But it _might_ be just a smidge clearer.

File: usng2-js/build.gradle.kts (hunk 2 above)
Comment (coyotesqrl) [reply usng2-js/build.gradle.kts:44]: For this project, I don't think it matters too much. For larger ones, it definitely will.

File: usng2-js/build.gradle.kts (hunk 3 above)
Comment (coyotesqrl) [reply usng2-js/build.gradle.kts]: 👍

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Pull request merged**